packages/ec-core/src/operator/selector/tournament.rs

```diff
@@ -1,15 +1,23 @@
 use std::num::NonZeroUsize;
 
-use anyhow::{ensure, Context, Result};
+use miette::Diagnostic;
 use rand::{prelude::IndexedRandom, rngs::ThreadRng};
 
 use super::Selector;
 use crate::population::Population;
 
+#[derive(Debug)]
 pub struct Tournament {
     size: NonZeroUsize,
 }
 
+#[derive(Debug, thiserror::Error, Diagnostic)]
+pub enum TournamentError {
+    #[error("Tournament size {0} was larger than population size {1}")]
+    #[diagnostic(help = "Ensure that the population has at least {0} individuals")]
+    LargerThanPopulation(usize, NonZeroUsize),
+}
+
```

Comment:
Since we don't have another error other than `LargerThanPopulation` and I don't see us changing the error paths of a TournamentSelector anytime soon, let's pull that out into a struct.